src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -365,10 +393,82 @@ public override BoundNode VisitInterpolatedString(BoundInterpolatedString node)
             return result;
         }
 
-        private BoundExpression LowerPartsToString(InterpolatedStringHandlerData data, ImmutableArray<BoundExpression> parts, SyntaxNode syntax, TypeSymbol type)
+        private (BoundExpression Result, bool RequiresVisitAndConversion) LowerPartsToConcatString(ImmutableArray<BoundExpression> parts, TypeSymbol type)
         {
+            BoundExpression? result = null;
+            // All fill-ins, if any, are strings, and none of them have alignment or format specifiers.
+            // We can lower to a more efficient string concatenation
+            // The normal pattern for lowering is to lower subtrees before the enclosing tree. However in this case
+            // we want to lower the entire concatenation so we get the optimizations done by that lowering (e.g. constant folding).
+
+            int length = parts.Length;
+            if (length == 0)
+            {
+                // $"" -> ""
+                return (_factory.StringLiteral(""), false);
+            }
+
+            result = null;
+            for (int i = 0; i < length; i++)
+            {
+                var part = parts[i];
+                if (part is BoundStringInsert fillin)
+                {
+                    // this is one of the filled-in expressions
+                    if (fillin.Value is { ConstantValueOpt: { IsChar: true, CharValue: var charLiteralValue } })
+                    {
+                        // Converts char to string because $"{'c'}" should be "c", not 'c'
+                        part = _factory.StringLiteral(charLiteralValue.ToString());
```

Comment:
This needs to be escaped. Please make sure we have tests with `{` and `}` characters that will break.